pyresample/bucket/__init__.py

```diff
@@ -209,27 +208,47 @@ def _call_pandas_groupby_statistics(self, scipy_method, data, fill_value=None, s
         # Calculate the min of the data falling to each bin
         out_size = self.target_area.size
 
-        # merge into one Dataframe
-        df = dd.concat([dd.from_dask_array(self.idxs), dd.from_dask_array(weights)],
-                       axis=1)
-        df.columns = ['x', 'values']
+        # create the output bins
+        bins = da.linspace(0, out_size-1, out_size).astype('int')
 
-        if scipy_method == 'min':
-            statistics = df.map_partitions(lambda part: part.groupby(
-                                           np.digitize(part.x,
-                                                       bins=np.linspace(0, out_size, out_size)
-                                                       )
-                                           )['values'].min())
+        # sort idxs and weights based on weights
+        if statistic_method == 'min':
+            order = np.argsort(weights)
+        elif statistic_method == 'max':
+            order = np.argsort(weights)[::-1]
 
-        elif scipy_method == 'max':
-            statistics = df.map_partitions(lambda part: part.groupby(
-                                           np.digitize(part.x,
-                                                       bins=np.linspace(0, out_size, out_size)
-                                                       )
-                                           )['values'].max())
+        @dask.delayed
+        def get_block_statistic(block, mask_buffer, idxs_sorted, weights_sorted):
```

Comment:
Writing this here now before I forget: this should be moved to the global scope (a global function) or a staticmethod. Dask has trouble serializing local/hidden functions for non-threaded schedulers (distributed).